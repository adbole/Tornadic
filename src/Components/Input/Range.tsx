import { css } from "@emotion/react";
import styled from "@emotion/styled";

import { vars } from "ts/StyleMixins";


const Track = css({
    borderRadius: vars.borderRadius,
    backgroundColor: "rgba(136, 136, 136, 0.5)",
});

const Thumb = css({
    WebkitAppearance: "none",
    appearance: "none",
    height: "10px",
    width: "10px",
    borderRadius: vars.borderRadius,
    backgroundColor: "#6498fa",
});

const Input = styled.input({
    appearance: "none",
    background: "transparent",
    cursor: "pointer",
    width: "100%",

    "&::-moz-range-track": [Track, { height: "10px" }],
    "&::-moz-range-thumb": [Thumb, { border: "none" }],
    "&::-webkit-slider-runnable-track": Track,
    "&::-webkit-slider-thumb": Thumb,
});

export default function Range(
    props: Omit<
        React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>,
        "type" | "children"
    >
) {
    return <Input type="range" {...props} />;
}
