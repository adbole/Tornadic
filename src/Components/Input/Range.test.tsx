import { render } from "@testing-library/react";

import Range from "./Range"


test("Matches snapshot", () => {
    const { container } = render(
        <Range
            min={0}
            max={100}
            value={50}
            step={1}
        />
    );

    expect(container).toMatchSnapshot();
})