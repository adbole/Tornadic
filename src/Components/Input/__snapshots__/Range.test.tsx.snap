// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches snapshot 1`] = `
.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: transparent;
  cursor: pointer;
  width: 100%;
}

.emotion-0::-moz-range-track {
  border-radius: var(--border-radius);
  background-color: rgba(136, 136, 136, 0.5);
  height: 10px;
}

.emotion-0::-moz-range-thumb {
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 10px;
  width: 10px;
  border-radius: var(--border-radius);
  background-color: #6498fa;
  border: none;
}

.emotion-0::-webkit-slider-runnable-track {
  border-radius: var(--border-radius);
  background-color: rgba(136, 136, 136, 0.5);
}

.emotion-0::-webkit-slider-thumb {
  -webkit-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 10px;
  width: 10px;
  border-radius: var(--border-radius);
  background-color: #6498fa;
}

<div>
  <input
    class="emotion-0 emotion-1"
    max="100"
    min="0"
    step="1"
    type="range"
    value="50"
  />
</div>
`;
